src/components/select-field/BaseSelectField.js

```diff
@@ -365,8 +368,23 @@ class BaseSelectField extends React.Component<Props, State> {
         return selectOptions;
     };
 
+    getSelectButtonWidth = (): null | number => {
+        if (!this.selectButtonRef) {
+            return 0;
+        }
+
+        const { width } = this.selectButtonRef.getBoundingClientRect();
+
+        return width;
+    };
+
+    setSelectButtonRef = setReferenceNode => node => {
```

Comment:
I think we're moving toward `React.createRef` rather than a callback style. Can we store `setReferenceNode` on the instance and call it in `componentDidMount`, instead? Or is this the prescribed way to get the reference ref per the Popper docs?

Reply:
I did it this way because Popper's `Reference` uses a render prop in which it passes in its own `refHandler` (https://github.com/FezVrasta/react-popper/blob/master/src/Reference.js#L35). Is there a way in which I can use `React.createRef` and also not have it overwritten by the `refHandler` given to me by Popper? @priyajeet @jstoffan